Sources/ProjectSpec/SpecValidation.swift

```diff
@@ -201,6 +201,10 @@ extension Project {
             if let action = scheme.run, let config = action.config, getConfig(config) == nil {
                 errors.append(.invalidSchemeConfig(scheme: scheme.name, config: config))
             }
+
+            let invalidTestPlans: [TestPlan] = scheme.test?.testPlans.filter { !(basePath + $0.path).exists } ?? []
```

Comment:
Idea: 
You could save a `map` loop when you combine the logic of `filter` and `map` in a `compactMap`

**META CODE**
```swift 
if let testPlans = scheme.test?.testPlans { // You can also do `??` if you like it more. but since the lines above and below use `if let...` we should go with it too. 
    let invalidTestPlans [ValidationError] = testPlans.compactMap { 
        !(basePath + $0.path).exists ? .invalidTestPlan($0) : nil 
    }
    errors.append(contentsOf: invalidTestPlans)
}
```
or a bit shorter
```swift 
if let testPlans = scheme.test?.testPlans {
     errors.append(contentsOf: testPlans.compactMap { 
        !(basePath + $0.path).exists ? .invalidTestPlan($0) : nil // Not sure if compiler will find `ValidationError` enum
    })
}
```

Reply:
I prefer my version of the code as it moves downwards instead of to the right (nesting), thus I find it more readable.

Loop performance should not be an issue here as this code is executed on macs with modern hardware and the lists will be very small.